python/tank_vendor/shotgun_authentication/shotgun_authenticator.py

```diff
@@ -0,0 +1,225 @@
+            self._defaults_manager.get_http_proxy(),
+            self._defaults_manager.is_host_fixed()
+        )
+        return self.create_session_user(
+            login=login, session_token=session_token,
+            host=host, http_proxy=self._defaults_manager.get_http_proxy()
+        )
+
+    def create_session_user(self, login, session_token=None, password=None, host=None, http_proxy=None):
+        """
+        Create an AuthenticatedUser given a set of human user credentials.
+        Either a password or session token must be supplied. If a password is supplied,
+        a session token will be generated for security reasons.
+
+        :param login: Shotgun user login
+        :param session_token: Shotgun session token
+        :param password: Shotgun password
+        :param host: Shotgun host to log in to. If None, the default host will be used.
+        :param http_proxy: Shotgun proxy to use. If None, the default http proxy will be used.
+
+        :returns: A SessionUser instance.
+        """
+        # Get the defaults is arguments were None.
+        host = host or self._defaults_manager.get_host()
+        http_proxy = http_proxy or self._defaults_manager.get_http_proxy()
+
+        # Create a session user
+        return user.ShotgunUser(
+            user_impl.SessionUser(host, login, session_token, http_proxy, password=password)
+        )
+
+    def create_script_user(self, api_script, api_key, host=None, http_proxy=None):
+        """
+        Create an AuthenticatedUser given a set of script credentials.
+
+        :param script_user: Shotgun script user
+        :param script_key: Shotgun script key
+        :param host: Shotgun host to log in to. If None, the default host will
+                     be used.
+        :param http_proxy: Shotgun proxy to use. If None, the default http proxy
+                           will be used.
+
+        :returns: A ShotgunUser derived instance.
+        """
+        return user.ShotgunUser(
+            user_impl.ScriptUser(
+                host or self._defaults_manager.get_host(),
+                api_script,
+                api_key,
+                http_proxy or self._defaults_manager.get_http_proxy(),
+            )
+        )
+
+    def get_default_host(self):
+        """
+        Returns the host from the defaults manager.
+        :returns: The default host string.
+        """
+        return self._defaults_manager.get_host()
+
+    def get_default_http_proxy(self):
+        """
+        Returns the HTTP proxy from the defaults manager.
+        :returns: The default http proxy string.
+        """
+        return self._defaults_manager.get_http_proxy()
+
+    def get_default_user(self):
+        """
+        Returns the default user from the defaults manager.
+        :returns: A ShotgunUser derived instance if available, None otherwise.
+        """
+        # Get the credentials
+        credentials = self._defaults_manager.get_user_credentials()
```

Comment:
I am still not sure I understand the meaning of this `defaults_manager.get_user_credentials()` - what is the intent? Can it return different things (e.g. script credentials / user credentials)? Docstring just says

```
        """
        Override to provide a default user's dictionary of credentials.

        :returns: Default implementation returns None.
        """
```

Curious about the intent here - wondering if we can tweak the form on which the data is being returned.